server/jetstream_cluster.go

```diff
@@ -2263,9 +2263,11 @@ func (js *jetStream) monitorStream(mset *stream, sa *streamAssignment, sendSnaps
 	// from underneath the one that is running since it will be the same raft node.
 	defer func() {
 		// We might be closing during shutdown, don't pre-emptively stop here since we'll still want to install snapshots.
+		mset.mu.RLock()
 		if !mset.closed.Load() {
```

Comment:
This is atomic, so don't need lock for that and since we already have `n` dereferenced not sure why these are needed?

Reply:
Removed it.